src/App.tsx

```diff
@@ -30,12 +31,20 @@ const carleton = {
 
 const fileChannel = 'file-content';
 const App: React.FC = () => {
-  const [data, setData] = useState<FeatureCollection[] | null>(null);
+  const [feature, setFeature] = useState<Feature[] | null>(null);
+
+  const displayFeature = useCallback(({ features }: MapEvent) => {
+    if (features) {
+      setFeature(features);
+    }
```

Comment:
Is there any specific reason for this check since `feature` can hold a `null` value?

Reply:
Yeah, it's kinda redundant ig

I was thinking, it'd be a waste of a re-render if a null value is passed in, but upon reflection, it makes sense to just update the feature LOL